component/host.go

```diff
@@ -23,8 +23,14 @@ type Host interface {
 	//
 	// ReportFatalError should be called by the component anytime after Component.Start() ends and
 	// before Component.Shutdown() begins.
+	// Deprecated: [0.65.0] Use ReportComponentStatus instead (with an event of type status.ComponentError)
 	ReportFatalError(err error)
 
+	// ReportComponentStatus can be used by a component to communicate its status to the Host.
+	// The Host implementations may broadcast this information to interested parties via
+	// StatusWatcher interface.
+	ReportComponentStatus(event *StatusEvent)
```

Comment:
Addition of this method is a breaking change. See failure on Jaeger repo: https://github.com/open-telemetry/opentelemetry-collector/actions/runs/3481700391/jobs/5823123326

Reply:
I submitted [a contrib PR](https://github.com/open-telemetry/opentelemetry-collector/pull/6553) that should remove the dependency on Jaeger collector, so we can make contrib tests pass, but I believe we will still likely need to run through a breaking change.

I see a few other problems with the current state of this interface:
1. The set of methods doesn't seem to be complete and potentially can change after we release 1.0, e.g. we might want to introduce getters for receivers and processors, not only exporters and extensions.
2. Most of the methods start with `Get` which goes against Go recommendations, maybe we want to change them and remove `Get` prefix.

I believe we need to take the opportunity and make a bigger refactoring that would allow us to solve all the outlined problems at once. We discussed the problem with @bogdandrutu and came up with the following plan:
Keep only one "core" method `ReportComponentStatus` the `Host` interface and move all other "getter" methods to optional interfaces. The optional interfaces for exporter and extension getters can be moved to their own packages (exporter/extension), so we don't need to change returned values from `Extension|Exporter` to`Component` as done in https://github.com/open-telemetry/opentelemetry-collector/pull/6553 anymore.

I'll play with it, submit a draft PR, then we can combine it with this PR and try to migrate to the new interfaces as gracefully as feasible.

Reply:
I'm catching up on this PR. It looks like this is where it stalled out. Some thoughts on how to move forward:

> The set of methods doesn't seem to be complete and potentially can change after we release 1.0, e.g. we might want to introduce getters for receivers and processors, not only exporters and extensions.

I'm proposing that we deprecate `GetExporters` (https://github.com/open-telemetry/opentelemetry-collector/issues/7370). I don't believe there are valid reasons to need this function or others like it, such as `GetReceivers`, `GetProcessors`, etc.

> Most of the methods start with Get which goes against Go recommendations, maybe we want to change them and remove Get prefix.

Between renaming `GetExtensions` to `Extensions`, deprecating `ReportFatalError` and `GetExporters`, and adding `ReportComponentStatus`, it looks like we want quite a different interface. Hard to believe we will just live with this as is, so ideally we just need to identify the best migration option. It looks like we no longer need to coordinate with Jaeger but presumably this will affect other components defined elsewhere.

I also tried to identify a graceful option. The best I came up with is quite ugly, but would require no unannounced breaking changes does not require changes to any other interfaces (i.e. `component.Component`):

1. Add an optional interface so it is possible to use the new functions. I'll call it `component.Host2`.
2. Deprecate `ReportFatalError`, `GetExporters`, and `GetExtensions` to alert users of these specific functions. Recommend the ugly solution of asserting the optional interfaces (temporarily).
3. After some time Deprecate `component.Host` to alert all all affected. However, clearly indicate that this interface will be _changed_  to `component.Host`, rather than removed.
4. Change `component.Host` to match `component.Host2`. Undeprecate `component.Host` and deprecate `component.Host2`